src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,377 @@
+  *Note: If  Visual Studio adds a default class named `Class1.cs`, feel free to delete this file to keep things tidy.
+6. For the fourth and final project, add another .NET Standard – Class Library project and name it **HelloWorld.Grains**.
+7. Inside the Grains project, add a class and name it **HelloGrain.cs**.
+
+The structure of your Solution should look like this:
+
+```
+
+Solution 'HelloWorld' (4 projects)
+> HelloWorld.Grains
+  > Dependencies
+  > HelloGrain.cs
+> HelloWorld.Interfaces
+  > Dependencies
+  > IHello.cs
+> OrleansClient
+  > Dependencies
+  > Program.cs
+> SiloHost
+  > Dependencies
+  > Program.cs
+
+```
+
+## Step Two: Add Orleans Packages using NuGet
+
+Right-click on each project and select **Manage NuGet Packages…**
+Search for and install the packages for each project, as listed here:
+
+SiloHost
+
+- Microsoft.Orleans.OrleansProviders
+- Microsoft.Orleans.OrleansRuntime
+- Microsoft.Extensions.Logging.Console
+
+OrleansClient
+
+- Microsoft.Orleans.Core
+- Microsoft.Extensions.Logging.Console
+
+HelloWorld.Grains
+
+- Microsoft.Orleans.Core.Abstractions
+- Microsoft.Orleans.OrleansCodeGenerator.Build
+- Microsoft.Extensions.Logging.Abstractions
+
+GrainInterfaces
+- Microsoft.Orleans.Core.Abstractions
+- Microsoft.Orleans.OrleansCodeGenerator.Build
+
+## Step Three: Configure project dependencies
+
+The SiloHost, OrleansClient, and HelloWorld.Grain projects each have dependencies on other projects in the solution.
+HelloWorld.Interfaces does not have any dependencies.
+
+SiloHost
+
+- HelloWorld.Grains
+- HelloWorld.Interfaces
+
+OrleansClient
+
+- HelloWorld.Interfaces
+
+HelloWorld.Grains
+
+- HelloWorld.Interfaces
+
+1. To add a dependency, right-click the project and select **Add…** -> **Reference…**.
+2. Check the boxes for the projects you need and click **OK.**
+
+## Step Four: Replace the default code
+
+### SiloHost - Program.cs
+
+```csharp
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using HelloWorld.Grains;
+using Microsoft.Extensions.Logging;
+using Orleans;
+using Orleans.Configuration;
+using Orleans.Hosting;
+
+namespace OrleansSiloHost
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            return RunMainAsync().Result;
+        }
+
+        private static async Task<int> RunMainAsync()
+        {
+            try
+            {
+                var host = await StartSilo();
+                Console.WriteLine("Press Enter to terminate...");
+                Console.ReadLine();
+
+                await host.StopAsync();
+
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return 1;
+            }
+        }
+
+        private static async Task<ISiloHost> StartSilo()
+        {
+            // define the cluster configuration
+            var builder = new SiloHostBuilder()
```

Comment:
no need to paste the whole file, as in it is in sample code as well. It is more important here to describe how to use SiloHostBuilder to configure a silo. So maybe just keep this block of code, and add annotation for it.

Reply:
Thanks!